### PR TITLE
Fail when using collections in Resteasy Reactive JAXB

### LINKED ISSUE #33865: RestEasy Reactive + JAXB unable to serialize list of entities
### Describe the bug

When using a call to the database with the IN operator no records found.

```java
MigrationResource.java

   /**
     * Find by customer id uni.
     *
     * @param customerId the customer id
     * @return the uni
     */
    @GET
    @Path("/customerId/multi")
    @Produces(MediaType.APPLICATION_XML)
    @Consumes(MediaType.TEXT_PLAIN)
    public Uni<List<Accounts>> findByCustomerId(@RestQuery @Separator(",") List<String> customerId) {
        return Accounts.findByCustomerId(customerId);
    }

...

Accounts.java

package se.hakuseki.ciam.migration;

import io.quarkus.hibernate.reactive.panache.PanacheEntity;
import io.quarkus.panache.common.Parameters;
import io.quarkus.runtime.annotations.RegisterForReflection;
import io.smallrye.mutiny.Uni;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
import lombok.ToString;

import java.time.LocalDateTime;
import java.time.OffsetDateTime;
import java.util.List;
import java.util.UUID;

// tag::Accounts[]

/**
 * The type Accounts.
 */
@Entity
@Cacheable
@Getter
@Setter
@ToString
@RegisterForReflection
@NamedQueries({
        @NamedQuery(name = "Accounts.getByCustomerId", query = "from Accounts where agasCustomerNumber = ?1", hints = @QueryHint(name = "org.hibernate.cacheable", value = "true"))
})
public class Accounts extends PanacheEntity {
    /**
     * The Uuid.
     */
    @Column(name = "pisa_uid", updatable = false)
    UUID pisaUid;
    /**
     * The User name.
     */
    @Column(name = "username", updatable = false)
    String userName;
    /**
     * The Person number.
     */
    @Column(name = "personnumber", updatable = false)
    String personNumber;
    /**
     * The First name.
     */
    @Column(name = "first_name", updatable = false)
    String firstName;
    /**
     * The Last name.
     */
    @Column(name = "last_name", updatable = false)
    String lastName;
    /**
     * The Email.
     */
    @Column(updatable = false)
    String email;
    /**
     * The Password.
     */
    @Column(updatable = false)
    String password;
    /**
     * The Customer number.
     */
    @Column(name = "customernumber", updatable = false)
    String customerNumber;
    /**
     * The Company customer number.
     */
    @Column(name = "companycustomernumber", updatable = false)
    String companyCustomerNumber;
    /**
     * The Organisation customer number.
     */
    @Column(name = "customernumberorganisation", updatable = false)
    String organisationCustomerNumber;
    /**
     * The Migration state.
     */
    @Column(name = "migration_state")
    boolean migrationState;
    /**
     * The Imported timestamp.
     */
    @Column(name = "imported_timestamp", updatable = false)
    LocalDateTime importedTimestamp;
    /**
     * The Migrated timestamp.
     */
    @Column(name = "migrated_timestamp")
    OffsetDateTime migratedTimestamp;

    /**
     * The Customer number ag.
     */
    @Column(name = "agas_customernumber", updatable = false)
    String agasCustomerNumber;

    /**
     * The Last login.
     */
    @Column(name = "last_login", updatable = false)
    LocalDateTime lastLogin;



    /**
     * Find by customer id uni.
     *
     * @param customerId the customer id
     * @return the uni
     */
    public static Uni<Accounts> findByCustomerId(String customerId) {
        return find("#Accounts.getByCustomerId", customerId).singleResult();
    }

    /**
     * Find by customer id uni.
     *
     * @param customerId the customer id
     * @return the uni
     */
    public static Uni<List<Accounts>> findByCustomerId(List<String> customerId) {
        return find("agasCustomerNumber in :vals ", Parameters.with("vals", customerId)).list();
    }
}
// end::Accounts[]

``` 

### Expected behavior

Records found

### Actual behavior

Empty response

### How to Reproduce?

Provided with a .http file

Five records in the database with ```agas_customernumber``` values from 1-5

### Output of `uname -a` or `ver`

Darwin Mikaels-MacBook-Pro.local 21.6.0 Darwin Kernel Version 21.6.0: Mon Apr 24 21:10:53 PDT 2023; root:xnu-8020.240.18.701.5~1/RELEASE_X86_64 x86_64

### Output of `java -version`

openjdk version "17.0.5" 2022-10-18 OpenJDK Runtime Environment GraalVM CE 22.3.0 (build 17.0.5+8-jvmci-22.3-b08) OpenJDK 64-Bit Server VM GraalVM CE 22.3.0 (build 17.0.5+8-jvmci-22.3-b08, mixed mode, sharing)

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

3.1.0.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.9.1 (2e178502fcdbffc201671fb2537d0cb4b4cc58f8)

### Additional information

Use this repo for tests

https://github.com/hakuseki/ciammigrering

### PR DESCRIPTION
**Pull request by Sgitario**
Fix https://github.com/quarkusio/quarkus/issues/33865

More details in https://github.com/quarkusio/quarkus/issues/33865#issuecomment-1594222208.

@geoand we could also validate also the inputs, so I'm not sure if we should do this, wdyt?

### CODE REVIEW

**Comment by geoand** [comment]
With regards to the question about the input:  what happens currently? Does it fail at runtime in a similar way as it does for the output?

**Comment by Sgitario** [comment]
@geoand PR updated

File: extensions/resteasy-reactive/quarkus-resteasy-reactive-jaxb/deployment/src/main/java/io/quarkus/resteasy/reactive/jaxb/deployment/ResteasyReactiveJaxbProcessor.java [hunk 1]
```diff
@@ -100,8 +111,16 @@ void registerClassesToBeBound(ResteasyReactiveResourceMethodEntriesBuildItem res
             boolean consumesXml = consumesXml(resourceInfo);
             boolean consumesMultipart = consumesMultipart(resourceInfo);
             if (consumesXml || consumesMultipart) {
-                for (Type parameter : methodInfo.parameterTypes()) {
-                    ClassInfo effectiveParameter = getEffectiveClassInfo(parameter, indexView);
+                for (MethodParameterInfo parameter : methodInfo.parameters()) {
+                    if (!isTypeCompatibleWithJaxb(parameter.type())) {
```
Comment (geoand) [inline extensions/resteasy-reactive/quarkus-resteasy-reactive-jaxb/deployment/src/main/java/io/quarkus/resteasy/reactive/jaxb/deployment/ResteasyReactiveJaxbProcessor.java:115]: Is this perhaps too general? I mean, by looking at the code it seems to not be limited to body parameters, but instead apply to all

Comment (Sgitario) [reply extensions/resteasy-reactive/quarkus-resteasy-reactive-jaxb/deployment/src/main/java/io/quarkus/resteasy/reactive/jaxb/deployment/ResteasyReactiveJaxbProcessor.java:115]: I know what you mean, and that's why I scratched the idea of doing it at first. 
Do you know if there is a utility or similar to know whether a parameter is an actual body parameter? One possible solution is to check the parameters that have no annotations. Yet it's not accurate enough.

Comment (geoand) [reply extensions/resteasy-reactive/quarkus-resteasy-reactive-jaxb/deployment/src/main/java/io/quarkus/resteasy/reactive/jaxb/deployment/ResteasyReactiveJaxbProcessor.java:115]: See https://github.com/quarkusio/quarkus/blob/158d4c6e39731b81b66c9553fe4b967c6bfadaf3/extensions/resteasy-reactive/quarkus-resteasy-reactive/deployment/src/main/java/io/quarkus/resteasy/reactive/server/deployment/QuarkusServerEndpointIndexer.java#L178

Comment (Sgitario) [reply extensions/resteasy-reactive/quarkus-resteasy-reactive-jaxb/deployment/src/main/java/io/quarkus/resteasy/reactive/jaxb/deployment/ResteasyReactiveJaxbProcessor.java]: PR updated using this.

File: independent-projects/resteasy-reactive/common/processor/src/main/java/org/jboss/resteasy/reactive/common/processor/EndpointIndexer.java [hunk 2]
```diff
@@ -1334,6 +1334,7 @@ && isParameterContainerType(paramType.asClassType())) {
                 if (field) {
                     return builder;
                 }
+                builder.setName(sourceName);
```
Comment (Sgitario) [inline independent-projects/resteasy-reactive/common/processor/src/main/java/org/jboss/resteasy/reactive/common/processor/EndpointIndexer.java:1337]: @geoand I had to add this to know the parameter name.

Comment (geoand) [reply independent-projects/resteasy-reactive/common/processor/src/main/java/org/jboss/resteasy/reactive/common/processor/EndpointIndexer.java:1337]: Makes sense

**Review by geoand — APPROVED** [review]

**Pull request merged**